plugin.php

```diff
@@ -289,9 +289,12 @@ function rest_validate_request_arg( $value, $request, $param ) {
 		$args = $attributes['args'][ $param ];
 
 		if ( ! empty( $args['enum'] ) ) {
-			if ( ! in_array( $value, $args['enum'] ) ) {
+			if ( 'string' === $args['type'] && ! in_array( $value, $args['enum'] ) ) {
 				return new WP_Error( 'rest_invalid_param', sprintf( __( '%s is not one of %s' ), $param, implode( ', ', $args['enum'] ) ) );
 			}
+			if ( 'array' === $args['type'] && array_diff( $value, $args['enum'] ) ) {
```

Comment:
Hmm this is somewhat new behaviour, as this is really "array of type", the valid values being the enums. We might be better of using the jsonSchema way of doing this, rather than inventing our own.